pkg/reconciler/reconciler.go

```diff
@@ -274,6 +276,17 @@ func SkipDependentWatches(skip bool) Option {
 	}
 }
 
+// StripManifestFromStatus is an Option that configures whether the manifest
+// should be removed from the automatically populated status.
+// This is recommended if the manifest might return sensitive data (i.e.,
+// secrets).
+func StripManifestFromStatus(strip bool) Option {
+	return func(r *Reconciler) error {
+		r.stripManifestFromStatus = strip
+		return nil
+	}
+}
```

Comment:
Rather than a specific "strip the manifest" manipulation, should we allow arbitrary manipulation of the release that is written to the status? Something like:

```
func WithReleaseStatusMutator(mutator func(*release.Release)) Option {
	return func(r *Reconciler) error {
		r.releaseStatusMutator = mutator
		return nil
	}
}
```